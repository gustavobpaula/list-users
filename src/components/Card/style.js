import styled from 'styled-components';

export const CardStyle = styled.article`
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 5px;
  cursor: pointer;
  font-family: Roboto, Arial;
  margin: 5px;
  max-width: 300px;
  padding: 20px 30px;
  transition: background-color .3s ease;

  &:hover {
    background-color: #eee;
  }
`;

export const NameStyle = styled.h3`
  margin: 0;
`;

export const CompanyStyle = styled.p`
  margin: 0;
`;
