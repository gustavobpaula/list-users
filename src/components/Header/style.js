import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const HeaderStyle = styled.header`
  border-bottom: 2px solid #ccc;
  margin-bottom: 30px;
  padding: 20px 0;
  text-align: center;
`;

export const LinkStyle = styled(Link)`
  color: #333;
  font-size: 50px;
  text-decoration: none;

  @media (max-width: 425px) {
    font-size: 30px;
  }
`;
