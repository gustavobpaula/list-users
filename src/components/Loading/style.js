import styled from 'styled-components';

export const LoadingStyle = styled.img`
  left: 50%;
  max-width: 100px;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;
