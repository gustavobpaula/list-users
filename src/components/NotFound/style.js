import styled from 'styled-components';

export const NotFoundStyle = styled.img`
  left: 50%;
  max-width: 100%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;
