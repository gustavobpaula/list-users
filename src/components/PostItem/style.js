import styled from 'styled-components';

export const Post = styled.article`
  border-bottom: 1px solid #ccc;
  display: grid;
  grid-gap: 50px;
  grid-template-columns: 1fr 1fr;
  padding: 80px 0;

  &:nth-child(even) {
    grid-auto-flow: dense
  }
`;

export const BoxTitle = styled.div`
  grid-column: 2;

  &.even {
    grid-column: 1;
  }
`;

export const BoxDescription = styled.div`
  grid-column: 1;

  &.even {
    grid-column: 2;
  }
`;

export const Title = styled.h1`
  color: #333;
  font-size: 45px;
  margin: 0;

  @media (max-width: 425px) {
    font-size: 25px;
  }
`;

export const User = styled.p`
  color: #999;
`;

export const Description = styled.p`
  line-height: 200%;
  margin: 0;
`;
