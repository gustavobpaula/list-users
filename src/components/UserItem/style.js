import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Card = styled.article`
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 10px;
  cursor: pointer;
  margin: 5px;
  max-width: 300px;
  padding: 20px 30px;
  transition: background-color .3s ease;

  &:hover {
    background-color: #eee;
  }
`;

export const Name = styled.h3`
  color: #000;
  margin: 0 0 20px 0;
  text-decoration: none;
`;

export const Company = styled.p`
  color: #999;
  margin: 0;
  text-decoration: none;
`;

export const StyledLink = styled(Link)`
  text-decoration: none;
`;
