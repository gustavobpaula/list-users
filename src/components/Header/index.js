import React from 'react';
import { HeaderStyle, LinkStyle } from './style';


export default function Header() {
  return (
    <HeaderStyle>
      <h1><LinkStyle to="/">Lorem Ipsum</LinkStyle></h1>
    </HeaderStyle>
  );
}
