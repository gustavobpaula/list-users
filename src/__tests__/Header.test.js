import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { LinkStyle } from '../components/Header/style';
import Header from '../components/Header';

describe('Testing Header Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Header />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should includes link to Home', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(LinkStyle).props().to).toBe('/');
  });
});
