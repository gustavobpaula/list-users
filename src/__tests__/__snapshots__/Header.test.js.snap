// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component should render correctly 1`] = `
<styled.header>
  <h1>
    <Styled(Link)
      to="/"
    >
      Lorem Ipsum
    </Styled(Link)>
  </h1>
</styled.header>
`;
